nucypher/characters/lawful.py

```diff
@@ -555,20 +555,23 @@ def generate_work_orders(self, map_id, *capsules, num_ursulas=None):
                 "Bob doesn't have a TreasureMap to match any of these capsules: {}".format(
                     capsules))
 
-        for node_id, arrangement_id in treasure_map_to_use:
-            ursula = self.known_nodes[node_id]
+        for ursula_checksum_address, arrangement_id in treasure_map_to_use:
+            ursula = self.known_nodes[ursula_checksum_address]
+            assert ursula.checksum_public_address == ursula_checksum_address
```

Comment:
Production `assert`? This will be the third one.  Perhaps a comment will help.

Reply:
Oops! My bad, I was just checking something and I forgot about it.